src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -138,27 +128,39 @@ public void processResponseFiles(int batchSize) {
         }
     }
 
-    private void processInputUploadBlob(String inputUploadBlob) {
+    private void processInputUploadBlob(String inputUploadBlob, UserAccountEntity userAccount) {
         log.debug("Found ARM Input Upload file {}", inputUploadBlob);
         try {
             BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor = new BatchInputUploadFileFilenameProcessor(inputUploadBlob);
             String manifestName = generateManifestName(batchUploadFileFilenameProcessor.getUuidString());
+
+            String inputUploadFileRecordStr = armDataManagementApi.getBlobData(inputUploadBlob).toString();
+            log.info("Contents of ARM Input Upload file: '{}' '{}", inputUploadBlob, inputUploadFileRecordStr);
+            ArmResponseInputUploadFileRecord inputUploadFileRecord = objectMapper.readValue(inputUploadFileRecordStr, ArmResponseInputUploadFileRecord.class);
             List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities = externalObjectDirectoryRepository
                 .findAllByStatusAndManifestFile(EodHelper.armDropZoneStatus(), manifestName);
+
             if (CollectionUtils.isNotEmpty(externalObjectDirectoryEntities)) {
+                OffsetDateTime timestamp = inputUploadFileRecord.getTimestamp();
```

Comment:
Based on my comments above, you will need to check with @hemantasharma1129 which timestamp you will need to use as there are multiples. Bare in mind the manifest file is written after the raw data is all uploaded so 24 hours later will mean the manifest will have inserted the first records before the last records so you may end up with some raw data that was ingested over 24 hours ago but not all the data in the manifest and until the manifest file is pushed, ARM cannot send response files. You might be best to use the last ingestion date for the EODS with the manifest file name unless there is a date field for when the manifest is pushed.

Reply:
See @hemantasharma1129 comment above